### PR TITLE
Adding new Resource Public maintenance configurations and GET/LISTAPI

### PR DESCRIPTION
**Pull request by sotaneja**
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/10281

### CODE REVIEW

**Comment by isra-fel** [comment]
@allenjzhang  could you take this one? Thanks

File: sdk/maintenance/Microsoft.Azure.Management.Maintenance/tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -20,5 +20,5 @@
 
 // The following GUID is for the ID of the typelib if this project is exposed to COM
 [assembly: Guid("6D17E66C-92BB-455C-A150-45922A0BA619")]
-[assembly: AssemblyVersion("1.1.0.0")]
-[assembly: AssemblyFileVersion("1.1.0.0")]
+[assembly: AssemblyVersion("1.2.0.0")]
```
Comment (isra-fel) [inline sdk/maintenance/Microsoft.Azure.Management.Maintenance/tests/Properties/AssemblyInfo.cs:23]: You should update the `AssemblyInfo.cs` in your `src/` not `tests/`. Test libraries are never published so their assemblyinfo need not be updated

**Review by isra-fel — CHANGES_REQUESTED** [review]
Please write tests for the new resource type. Thanks

File: sdk/maintenance/Microsoft.Azure.Management.Maintenance/tests/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (sotaneja) [reply sdk/maintenance/Microsoft.Azure.Management.Maintenance/tests/Properties/AssemblyInfo.cs:23]: Seems we were missing the assemblyinfo, thanks for catching

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**